packages/core/src/baseCogniteClient.ts

```diff
@@ -401,6 +447,70 @@ export default class BaseCogniteClient {
   protected get httpClient() {
     return this.http;
   }
+
+  protected loginWithAAD = ({
+    cluster,
+    clientId,
+    tenantId,
+    signInType,
+    debug,
+  }: OAuthLoginForAADOptions) => {
+    const config = {
+      auth: {
+        clientId,
+        authority: `https://login.microsoftonline.com/${tenantId}`,
+        redirectUri: `${window.location.origin}`,
+      },
+    };
+
+    const azureAdClient = new AzureAD({ config, cluster, debug });
+
+    this.azureAdClient = azureAdClient;
+
+    return async () => {
+      const account = await azureAdClient.initAuth();
+      const authResult = await azureAdClient.getProfileTokenRedirect();
```

Comment:
This doesn't look right, you are getting a token via redirect and doing a login (new redirect) below?

Reply:
yeah, make sense, completely redundant here